qutebrowser/browser/inspector.py

```diff
@@ -52,18 +54,50 @@ class WebInspectorError(Exception):
 
 class AbstractWebInspector(QWidget):
 
-    """A customized WebInspector which stores its geometry."""
+    """A customized WebInspector which stores its geometry.
 
-    def __init__(self, parent=None):
+    Attributes:
+        _position: position of the inspector (right/left/top/bottom/window)
+        _splitter: InspectorSplitter where the inspector can be placed.
+
+    Signals:
+        closed: Emitted when the inspector is closed.
+    """
+
+    closed = pyqtSignal()
+    _position = None
+    _splitter = None
+
+    def __init__(self, splitter, parent=None):
         super().__init__(parent)
         self._widget = None
         self._layout = miscwidgets.WrapperLayout(self)
-        self._load_state_geometry()
+        self._splitter = splitter
```

Comment:
This leaves the object in an "invalid" state until the first time `set_position` is called with argument not `None`. May be a problem.